### PR TITLE
wrf: Add New version, support fujitsu compiler, refactor and bug fix.

### PR DESCRIPTION
**Pull request by tkameyama**
The PR change to wrf as follows:

- Support Fujitsu compiler
- Add version 4.3.1 to 4.3.3
- output build log after build is ended
- fix external/io_netcdf
- Add WRF_HOME environment variable and PATH to prefix/{main,tools} when spack load wrf
- refactor to  change configure.defaults

### CODE REVIEW

File: var/spack/repos/builtin/packages/wrf/package.py [hunk 1]
```diff
@@ -381,6 +360,7 @@ def run_compile_script(self):
             error=str
         )
 
+        print(result_buf)
```
Comment (MichaelLaufer) [inline var/spack/repos/builtin/packages/wrf/package.py:363]: @adamjstewart @alalazo Can you confirm that  this is the "Spack" accepted way to output the string buffer to the build logs (and also to the screen when verbose mode is called)?
I feel like it should be through some functionality in llnl.util.tty.

**Background**
In order to check that the build has built successfully we need to parse the output of the compilation for a completion message. But the current implementation would discard this buffer, so the user could not inspect the errors.

**Comment by MichaelLaufer** [comment]
As I do not have access to other compilers except for Gnu, I would like to verify usability for the other compilers before we integrate these changes.
@amd-toolchain-support @mbabu-amd can you please confirm that the changes here work with aocc compiler?
@jedbrown can you confirm the same for Intel compiler (Re.  #28382)?

**Comment by jedbrown** [comment]
I'm guessing you mean to tag @t-brown. :point_up:

File: var/spack/repos/builtin/packages/wrf/package.py (hunk 1 above)
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/wrf/package.py:363]: Both tty and print are valid. Unfortunately there isn't a way to log to the screen when verbose is disabled. There have been several proposals to fix this but none of them were accepted.

**Comment by t-brown** [comment]
Confirmed, changes work with Intel.
@tkameyama thanks!

File: var/spack/repos/builtin/packages/wrf/package.py (hunk 1 above)
Comment (MichaelLaufer) [reply var/spack/repos/builtin/packages/wrf/package.py:363]: Thanks, so we will stay with this implementation

**Comment by MichaelLaufer** [comment]
@tkameyama I think this is great work, simplifies the package.py and adds extra functionality. Thanks! 
Regarding support for Fujitsu compilers in future WRF versions, can you contribute the configure stanza to WRF upstream? This will save us future effort maintaining these patches.

**Review by MichaelLaufer — APPROVED** [review]

**Review by tgamblin — APPROVED** [review]

**Comment by tgamblin** [comment]
Thanks everyone!

**Pull request merged**